File: framework/web/src/main/java/cn/bestwu/simpleframework/config/CorsConfiguration.java
package cn.bestwu.simpleframework.config;

import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.cors.CorsConfigurationSource;
import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
import org.springframework.web.filter.CorsFilter;

@ConditionalOnProperty(prefix = "security.cors", value = "enable", havingValue = "true")
@Configuration
@EnableConfigurationProperties(CorsProperties.class)
public class CorsConfiguration {

  @Bean("corsConfigurationSource")
  public CorsConfigurationSource corsConfigurationSource(CorsProperties corsProperties) {
    final UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
    source.registerCorsConfiguration(corsProperties.getPath(), corsProperties);
    return source;
  }

  @Bean
  public CorsFilter corsFilter(@Qualifier("corsConfigurationSource") CorsConfigurationSource configurationSource){
    return new CorsFilter(configurationSource);
  }

}

File: web/src/main/kotlin/top/bettercode/summer/web/config/CorsConfiguration.kt
package top.bettercode.summer.web.config

import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.boot.autoconfigure.condition.ConditionalOnClass
import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.Ordered
import org.springframework.core.annotation.Order
import org.springframework.web.cors.CorsConfigurationSource
import org.springframework.web.cors.UrlBasedCorsConfigurationSource
import org.springframework.web.filter.CorsFilter
import top.bettercode.summer.web.properties.CorsProperties

@ConditionalOnClass(javax.servlet.Filter::class)
@ConditionalOnProperty(prefix = "summer.security.cors", value = ["enable"], havingValue = "true", matchIfMissing = true)
@Configuration(proxyBeanMethods = false)
@ConditionalOnMissingBean(CorsFilter::class)
@EnableConfigurationProperties(CorsProperties::class)
class CorsConfiguration {
    @Bean("corsConfigurationSource")
    fun corsConfigurationSource(corsProperties: CorsProperties): CorsConfigurationSource {
        val source = UrlBasedCorsConfigurationSource()
        source.registerCorsConfiguration(corsProperties.path, corsProperties)
        return source
    }

    @Order(Ordered.HIGHEST_PRECEDENCE)
    @Bean
    fun corsFilter(
            @Qualifier("corsConfigurationSource") configurationSource: CorsConfigurationSource): CorsFilter {
        return CorsFilter(configurationSource)
    }
}
